NPE with new Parse initializer

From https://github.com/ParsePlatform/Parse-SDK-Android/commit/7cec3b41e4f96dc43b4d97e4783b03967aed41f0#commitcomment-14948582:

There seems to be a bug here...if I'm  using Stetho the builder will be not be instantitated and this.interceptors appears to be a null object.  I was able to workaround by downgrading to 1.11.0:

```java
Parse.addParseNetworkInterceptor(new ParseStethoInterceptor());
Parse.initialize(context, parseAppId, parseClientKey);
```

```
E/AndroidRuntime( 3595): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void java.util.List.clear()' on a null object reference
E/AndroidRuntime( 3595): 	at com.parse.Parse$Configuration$Builder.setNetworkInterceptors(Parse.java:157)
E/AndroidRuntime( 3595): 	at com.parse.Parse$Configuration$Builder.access$600(Parse.java:47)
E/AndroidRuntime( 3595): 	at com.parse.Parse.initialize(Parse.java:339)
```

--- a/Parse/src/main/java/com/parse/Parse.java
+++ b/Parse/src/main/java/com/parse/Parse.java
@@ -152,9 +152,14 @@ public class Parse {
         return this;
       }
 
-      private Builder setNetworkInterceptors(Collection<ParseNetworkInterceptor> interceptors) {
-        if (interceptors != null) {
+      /* package for tests */ Builder setNetworkInterceptors(Collection<ParseNetworkInterceptor> interceptors) {
+        if (this.interceptors == null) {
+          this.interceptors = new ArrayList<>();
+        } else {
           this.interceptors.clear();
+        }
+
+        if (interceptors != null) {
           this.interceptors.addAll(interceptors);
         }
         return this;

--- a/Parse/src/test/java/com/parse/ParseClientConfigurationTest.java
+++ b/Parse/src/test/java/com/parse/ParseClientConfigurationTest.java
@@ -12,6 +12,10 @@ import com.parse.http.ParseNetworkInterceptor;
 
 import org.junit.Test;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNull;
@@ -56,5 +60,61 @@ public class ParseClientConfigurationTest {
     } catch (UnsupportedOperationException ex) {
       // Expected
     }
+  }
+
+  @Test
+  public void testSetNetworkInterceptors() {
+    final ParseNetworkInterceptor interceptorA = mock(ParseNetworkInterceptor.class);
+    final ParseNetworkInterceptor interceptorB = mock(ParseNetworkInterceptor.class);
+
+    Collection<ParseNetworkInterceptor> collectionA = new ArrayList<ParseNetworkInterceptor>() {{
+      add(interceptorA);
+      add(interceptorB);
+    }};
+
+    Collection<ParseNetworkInterceptor> collectionB = new ArrayList<ParseNetworkInterceptor>() {{
+      add(interceptorB);
+      add(interceptorA);
+    }};
+
+    Parse.Configuration.Builder builder = new Parse.Configuration.Builder(null);
+
+    builder.setNetworkInterceptors(collectionA);
+    Parse.Configuration configurationA = builder.build();
+
+    builder.setNetworkInterceptors(collectionB);
+    Parse.Configuration configurationB = builder.build();
+
+    assertTrue(collectionsEqual(configurationA.interceptors, collectionA));
+    assertTrue(collectionsEqual(configurationB.interceptors, collectionB));
+  }
+
+  private static <T> boolean collectionsEqual(Collection<T> a, Collection<T> b) {
+    if (a.size() != b.size()) {
+      return false;
+    }
+
+    Iterator<T> iteratorA = a.iterator();
+    Iterator<T> iteratorB = b.iterator();
+    for (; iteratorA.hasNext() && iteratorB.hasNext();) {
+      T objectA = iteratorA.next();
+      T objectB = iteratorB.next();
+
+      if (objectA == null || objectB == null) {
+        if (objectA != objectB) {
+          return false;
+        }
+        continue;
+      }
+
+      if (!objectA.equals(objectB)) {
+        return false;
+      }
+    }
+
+    if (iteratorA.hasNext() || iteratorB.hasNext()) {
+      return false;
+    }
+    return true;
   }
 }

